tools/shellcheck-all.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -euo pipefail
+
+cd "$(git rev-parse --show-toplevel)"
+
+failed=""
+while IFS="" read -r file
+do
+    if grep -Eq '^#!(.*/|.*env +)(sh|bash|ksh)' "$file" || (echo "$file" | grep -Eq '.*.(ba)?sh$')
+    then
+        if ! shellcheck "$file"
+        then
+            failed=failed
```

Comment:
Any reason we can't just put an `exit 1` here, instead of having this variable? (My bash is a bit rusty, so I suspect the answer is something like "this part technically runs in a subshell" or something.)

Reply:
This is so that you see all errors, rather than only being able to see errors in later files after you've fixed earlier ones.